src/main/java/com/salesforce/dataloader/process/ProcessRunner.java

```diff
@@ -294,17 +308,37 @@ public static ProcessRunner getInstance(Map<String, String> argMap) throws Proce
         }
         logger = LogManager.getLogger(ProcessRunner.class);
         
-        if(argMap != null && argMap.containsKey(PROCESS_NAME)) {
-            // if process name is specified, get it from configuration
-            String processName = argMap.get(PROCESS_NAME);
-            runner = ProcessConfig.getProcessInstance(processName);
-            // command line param values override the ones in config file
-            runner.getConfigOverrideMap().putAll(argMap);
-        } else {
-            // if process.name is not given, load defaults & override with command-line args
-            runner = new ProcessRunner();
-            runner.setConfigOverrideMap(argMap);
+        try {
+            // get a controller instance to load the properties except for the
+            // runtime properties stored in XXX_lastRun.properties file.
+            Controller controller = Controller.getInstance("", true, null);
+            logger.info(Messages.getString("Process.initializingEngine")); //$NON-NLS-1$
+            Config config = controller.getConfig();
+            // load parameter overrides (from command line or caller context)
+            logger.info(Messages.getString("Process.loadingParameters")); //$NON-NLS-1$
+            config.loadParameterOverrides(argMap);
+            String processName = config.getString(PROCESS_NAME);
+            logger.debug("process name is " + processName);
+            if (processName == null || processName.isEmpty()) {
+                logger.info(PROCESS_NAME + "is not set in the command line or config.properties file.");
+                // operation and other process params are specified through properties
+                if (!validateConfigProperties(config)) {
+                    logger.fatal(Messages.getFormattedString("Process.missingRequiredArg", PROCESS_NAME));
+                    throw new ParameterLoadException(Messages.getFormattedString("Process.missingRequiredArg", PROCESS_NAME));
+                } else {
+                    runner = new ProcessRunner();
+                    runner.setName(config.getString(Config.OPERATION));
+                    runner.setConfigOverrideMap(argMap);
+                }
+            } else {
+                // process DynaBean name specified.
+                runner = ProcessConfig.getProcessInstance(processName);
+                runner.getConfigOverrideMap().putAll(argMap);
+            }
+        } catch (Exception e) {
+            throw new RuntimeException(e);
```

Comment:
Current method declares that it throws ProcessInitializationException, but looks like it will no longer be happening because we swallow all exceptions inside this clause and throw RuntimeException up the chain. Should this be 2 clauses? one to catch and re-throw ProcessInitializationException and the other one to throw new RuntimeException()

Also looking through this try/catch what else do we expect to throw an 'Exception' which we need to catch, why do we need this clause?

Reply:
Good catch! Removing try/catch block makes the most sense.

Reply:
updated the code.